### PR TITLE
Amend 2 TRD CalROC and CalDet to have name and title in spectra.

### PR DESCRIPTION
**Pull request by bazinski**
- Fix the copyright, remove the old one still there.
- ~~remove c style arrays and replace with vectors, leveraging the initialisation on instantiation~~.
- add back the titles and names for the root histograms as they are needed for at least debugging purposes, maybe more. Originally pulled in from TNamed which we dont inherit from anymore so a few hoops to jump through.

### CODE REVIEW

**Comment by bazinski** [comment]
@jolopezl sorry just saw your message now, weird to do a pull request and have a message sitting in it for 3 days ? how does that work. Been doing storage tenders and I presume I missed the mail, chat in meeting shortly.

**Comment by jolopezl** [comment]
So, no worries now. After todays discussion the meet gin we have the right idea behind this.

**Comment by sawenzel** [comment]
There are conflicts with dev. Could you please resolve them? Please also consider making a single or few clean commits for this PR.

**Comment by bazinski** [comment]
@sawenzel the codacy failures were not there on the first submission, scope fixed now and pushed. 
The o2-dev-fairroot stuff was related to not finding geant4 ( I think). 
Hopefully works now.

**Comment by jolopezl** [comment]
Hello @bazinski, I think you are leaving some code duplicated here. What is the reason of creating CatDet and CalRoc instead of moving TRDCalDet and TRDCalRoc? Maybe by using `git mv` to do the renaming?. That should do the work, and then we can actually see the difference before/after too.

**Comment by bazinski** [comment]
I left the TRDCalDet and TRDCalROC, because you asked me to leave them in so current code will work.
I can remove trdcaldet and trdcalroc and translate the places that use trdcaldet and trdcalroc?

**Comment by jolopezl** [comment]
@bazinski, I think we miscommunicated. There is no need to keep them both. It's better to apply the changes in TRDCal*, then do `git mv TRDCal* Cal*`, and get the history of the changes complete. To let the full chain working I think you would just need to change a couple of occurrences the Digitizer. I can rebase my developing branch after that, so no problem.

**Comment by bazinski** [comment]
@jolopezl cool, will fix after this last push finishes.

File: Detectors/TRD/base/src/CalDet.cxx [hunk 1]
```diff
@@ -560,28 +543,25 @@ double TRDCalDet::calcMean(bool wghtPads)
...
   double meanALL = 0.0;
   double meanWP = 0.0;
-  double pads = 0.0;
   double padsALL = (144 * 16 * 24 + 144 * 12 * 6) * 18;
-  std::array<double, 18> meanSM{};
-  std::array<double, 18> meanSMWP{};
-  std::for_each(meanSM.begin(), meanSM.end(), [](double& n) { n = 0; });
-  std::for_each(meanSMWP.begin(), meanSMWP.end(), [](double& n) { n = 0; });
+  std::vector<double> meanSM(18);
+  std::vector<double> meanSMWP(18);
```
Comment (jolopezl) [inline Detectors/TRD/base/src/CalDet.cxx]: Hi @bazinski,

I don't see the reason to use something beyond `std::array` for the definition of these quantities since they are fixed in size, thus generally faster. There are other places where I made this choice too, for example L74 in Detectors/TRD/base/include/TRDBase/CalDet.h.

I would agree to remove the two following lines `std::for_each` since they are not really necessary because array initializes all values to zero. Maybe `std::fill` would also ensure this too, but I am not sure about its performance compared to the `std::for_each` loop.

Comment (bazinski) [reply Detectors/TRD/base/src/CalDet.cxx]: Hi @jolopezl the reason for vector is that is zeros is internal data structure unlike array, besides that, if you dont cause it to expand its the same as an array performance wise.
Also fill is only optimised for char, the rest is a element wise walk through the structure.
So *besides* the boolean case, its faster to use memset to reset an array or vector.
There are examples in array??? of me using memset.
Maybe future versions of gcc and lvm will optimise the fill for basic types, but its not the case at the moment, just char.

Enjoy Bergen

Comment (sawenzel) [reply Detectors/TRD/base/src/CalDet.cxx]: not precisely. array has its memory in place. vector will allocate somewhere else. array should be more cache friendly and you can serialize the objects much better.
You can init arrays using `std::array<double,10> a{}` to default init!

Comment (jolopezl) [reply Detectors/TRD/base/src/CalDet.cxx]: No, we are not comparing `std::vector` with C-like arrays, but with `std::array` which is a template for a static size array, so it is initialized on the stack, while `std::vector` elements are allocated on the heap, causing an overhead.

As far as we have evolved in the code we have tried to optimize little things everywhere. So I guess there is no loss in doing the right choice by reverting just two lines.

Comment (bazinski) [reply Detectors/TRD/base/src/CalDet.cxx]: @sawenzel you mean stack vs heap? 
When I did a trivial test std::array does not intiialise its structure though.
mmmm I might go change a few things in other classes. I never bothered to check whether they ever *actually* change size after construction.

Comment (bazinski) [reply Detectors/TRD/base/src/CalDet.cxx]: @jolopezl sorry I replied while you were reply, ja I follow. Let me run along and convince myself ....

Comment (sawenzel) [reply Detectors/TRD/base/src/CalDet.cxx]: `std::array<double,10> a{}`  **NOTE THE `{}`** will init.

Comment (bazinski) [reply Detectors/TRD/base/src/CalDet.cxx]: wow capitals mmmm let me change asap ....  ;-)

**Comment by sawenzel** [comment]
@bazinski : I would like to go ahead but there is a merge conflict with dev. Could you rebase on dev?
Ideally, please also make a single authoritative commit with a summary of the developments (otherwise I plan to squash this).

**Comment by bazinski** [comment]
@sawenzel mmmm merge conflict looks nasty, will fix this evening.

**Comment by shahor02** [comment]
@bazinski it seems you need first to rebase your branch to dev ...

**Comment by bazinski** [comment]
@shahor02 I thought I did :( let me try again, else I will redo this PR

**Comment by bazinski** [comment]
@shahor02 ok that should be nice and clean with only 2 commit messages. Assuming the ci environment is happy, this should be good to go now. I hope ....
